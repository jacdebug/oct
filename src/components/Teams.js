import React from 'react';
import styled from 'styled-components';
import background from '../assets/background.png';

const Teams = ({ home, away }) =>
  <TeamsWraper>
    <Team data={home} />
    <TeamVs>Vs</TeamVs>
    <Team data={away} />
  </TeamsWraper>;

export default Teams;

const TeamsWraper = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;

  min-height: 150px;
  background-image: url(${background});
`;

const Team = ({ data }) =>
  <div>
    <TeamImg src={`/assets/${data.logo}`} />
    <TeamName>
      {data.name}
    </TeamName>
  </div>;

const TeamImg = styled.img`max-width: 100px;`;
const TeamName = styled.div`text-align: center`;
const TeamVs = styled.div`margin: 30px;`;
