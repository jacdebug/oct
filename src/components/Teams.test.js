import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Teams from './Teams';

const home = {
  name: 'Barcelona',
  logo: 'barcelona.png',
};

const away = {
  name: 'Granada',
  logo: 'granada.png',
};

it('renders without crashing', () => {
  shallow(<Teams home={home} away={away} />);
});

it('renders a snapshot', () => {
  const tree = shallow(<Teams home={home} away={away} />);
  expect(toJson(tree)).toMatchSnapshot();
});
