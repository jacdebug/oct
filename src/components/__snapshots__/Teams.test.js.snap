// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<styled.div>
  <Team
    data={
      Object {
        "logo": "barcelona.png",
        "name": "Barcelona",
      }
    }
  />
  <styled.div>
    Vs
  </styled.div>
  <Team
    data={
      Object {
        "logo": "granada.png",
        "name": "Granada",
      }
    }
  />
</styled.div>
`;
